webapp/src/main/java/com/box/l10n/mojito/android/strings/AndroidStringDocumentMapper.java

```diff
@@ -0,0 +1,186 @@
+        Map<String, AndroidPlural.AndroidPluralBuilder> pluralByOther = new HashMap<>();
+
+        textUnits.forEach(textUnit -> {
+
+            if (isSingularTextUnit(textUnit)) {
+                document.addSingular(textUnitToAndroidSingular(textUnit, useSource));
+            } else {
+                pluralByOther.compute(textUnit.getPluralFormOther(), (key, builder) -> {
+                    if (Objects.isNull(builder)) {
+                        builder = AndroidPlural.builder();
+                    }
+
+                    if (textUnit.getPluralForm().equalsIgnoreCase(OTHER.name())) {
+                        String name = PluralNameParser.getPrefix(textUnit.getName(), pluralSeparator);
+                        builder.setName(textUnit.getAssetPath() + assetDelimiter + name);
+                        builder.setComment(textUnit.getComment());
+                    }
+
+                    builder.addItem(new AndroidPluralItem(
+                            textUnit.getPluralForm(),
+                            textUnit.getTmTextUnitId(),
+                            useSource ? textUnit.getSource() : textUnit.getTarget()));
+
+                    return builder;
+
+                });
+            }
+        });
+
+        pluralByOther.forEach((pluralFormOther, builder) -> document.addPlural(builder.build()));
+
+        return document;
+    }
+
+    public AndroidStringDocument readFromSourceTextUnits(List<TextUnitDTO> textUnits) {
+        return readFromTextUnits(textUnits, true);
+    }
+
+    public AndroidStringDocument readFromTargetTextUnits(List<TextUnitDTO> textUnits) {
+        return readFromTextUnits(textUnits, false);
+    }
+
+    public List<TextUnitDTO> mapToTextUnits(AndroidStringDocument document) {
+        return mapToTextUnitStream(document).collect(Collectors.toList());
+    }
+
+    public Stream<TextUnitDTO> mapToTextUnitStream(AndroidStringDocument document) {
+        return document.getStrings()
+                .stream()
+                .flatMap(this::stringToTextUnits)
+                .map(this::addTextUnitDTOAttributes);
+    }
+
+    TextUnitDTO addTextUnitDTOAttributes(TextUnitDTO textUnit) {
+
+        if (!Strings.isNullOrEmpty(locale)){
+            textUnit.setTargetLocale(locale);
+        }
+
+        if (!Strings.isNullOrEmpty(repositoryName)){
+            textUnit.setRepositoryName(repositoryName);
+        }
+
+        if (textUnit.getName().contains(assetDelimiter)){
+            String[] nameParts = textUnit.getName().split(assetDelimiter);
+
+            if (nameParts.length == 2) {
+                textUnit.setAssetPath(nameParts[0]);
+                textUnit.setName(nameParts[1]);
+            }
+        }
+
+        return textUnit;
+    }
+
+    Stream<TextUnitDTO> stringToTextUnits(AbstractAndroidString androidString) {
+        if (androidString.isSingular()){
+            return singularToTextUnit((AndroidSingular) androidString);
```

Comment:
we've been trying to avoid multiple return in the code base in general. can use ternary operator or a variable

Reply:
updated